### DISCUSSION TITLE
Extension quarkus-azure-functions-http breaks event loop

### DISCUSSION THREAD

**Discussion by rumfuddle** [quarkusio/quarkus#23682] [Community]
Hi,

Don't know if this is expected behaviour but I lost a lot of time investigating this, so I thought it was best to report this.
I'm developing a small Quarkus demo app which should be deployed as an Azure function. Nothing fancy, just a RESTeasy Jackson API. Initially, without adding the Azure extension, the app runs fine locally, both in dev mode and as an uber jar.
But when I include the Azure extension, dev mode still works but as a jar, it doesn't start listening on its port. Looking at the debug log, I can see vert.x event loop activity in dev mode (an event loop thread is doing its work), but when the jar is used, there is no listener and everything is handled by the main thread.
I'm aware that the ultimate goal is to deploy it to Azure and if it works there, it should be ok, but I would still expect that the jar works locally as well, no?

**Comment by rquinio** [comment]
Another option could be to not generate the Quarkus Main in the jar when quarkus-azure-functions-http is present, especially if it's broken.
Indeed on Azure the Java main is managed by the Azure function and Quarkus class is called directly by reflection.
